volume-providers/vpc/vpcclient/vpcvolume/get_volume_test.go

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ * IBM Confidential
+ * OCO Source Materials
+ * IBM Cloud Container Service, 5737-D43
+ * (C) Copyright IBM Corp. 2018 All Rights Reserved.
+ * The source code for this program is not  published or otherwise divested of
+ * its trade secrets, irrespective of what has been deposited with
+ * the U.S. Copyright Office.
+ ******************************************************************************/
+
+package vpcvolume_test
+
+import (
+	"github.com/IBM/ibmcloud-storage-volume-lib/volume-providers/vpc/vpcclient/models"
+	"github.com/IBM/ibmcloud-storage-volume-lib/volume-providers/vpc/vpcclient/riaas/test"
+	"github.com/IBM/ibmcloud-storage-volume-lib/volume-providers/vpc/vpcclient/vpcvolume"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"net/http"
+	"os"
+	"testing"
+)
+
+func GetTestContextLogger() (*zap.Logger, zap.AtomicLevel) {
+	consoleDebugging := zapcore.Lock(os.Stdout)
+	consoleErrors := zapcore.Lock(os.Stderr)
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.TimeKey = "ts"
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	traceLevel := zap.NewAtomicLevel()
+	traceLevel.SetLevel(zap.InfoLevel)
+	core := zapcore.NewTee(
+		zapcore.NewCore(zapcore.NewJSONEncoder(encoderConfig), consoleDebugging, zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+			return (lvl >= traceLevel.Level()) && (lvl < zapcore.ErrorLevel)
+		})),
+		zapcore.NewCore(zapcore.NewJSONEncoder(encoderConfig), consoleErrors, zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+			return lvl >= zapcore.ErrorLevel
+		})),
+	)
+	logger := zap.New(core, zap.AddCaller())
+	return logger, traceLevel
+}
+
+func TestGetVolume(t *testing.T) {
+	// Setup new style zap logger
+	logger, _ := GetTestContextLogger()
+	defer logger.Sync()
+
+	testCases := []struct {
+		name string
+
+		// Response
+		status  int
+		content string
+
+		// Expected return
+		expectErr string
+		verify    func(*testing.T, *models.Volume, error)
+	}{
+		{
+			name:   "Verify that the correct endpoint is invoked",
+			status: http.StatusNoContent,
+		}, {
+			name:      "Verify that a 404 is returned to the caller",
+			status:    http.StatusNotFound,
+			content:   "{\"errors\":[{\"message\":\"testerr\"}]}",
+			expectErr: "Trace Code:, testerr Please check ",
+		}, {
+			name:    "Verify that the volume is parsed correctly",
+			status:  http.StatusOK,
+			content: "{\"id\":\"vol1\",\"status\":\"pending\"}",
+			verify: func(t *testing.T, volume *models.Volume, err error) {
+
+				if assert.NotNil(t, volume) {
+					assert.Equal(t, "vol1", volume.ID)
+				}
+			},
+		},
```

Comment:
Looks you are planing to add some more negative tests like across resource group and wrong a/c and wrong volume id etc.

Reply:
SURE

Reply:
yeah..working on fakes for that